test/integration/operator/extension/care/care_test.go

```diff
@@ -0,0 +1,426 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package care_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	operatorv1alpha1 "github.com/gardener/gardener/pkg/apis/operator/v1alpha1"
+	resourcesv1alpha1 "github.com/gardener/gardener/pkg/apis/resources/v1alpha1"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
+)
+
+var _ = Describe("Extension Care controller tests", func() {
+	var (
+		garden    *operatorv1alpha1.Garden
+		extension *operatorv1alpha1.Extension
+	)
+
+	BeforeEach(func() {
+		garden = &operatorv1alpha1.Garden{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   gardenName,
+				Labels: map[string]string{testID: testRunID},
+			},
+			Spec: operatorv1alpha1.GardenSpec{
+				RuntimeCluster: operatorv1alpha1.RuntimeCluster{
+					Networking: operatorv1alpha1.RuntimeNetworking{
+						Pods:     []string{"10.1.0.0/16"},
+						Services: []string{"10.2.0.0/16"},
+					},
+					Ingress: operatorv1alpha1.Ingress{
+						Domains: []operatorv1alpha1.DNSDomain{{Name: "ingress.runtime-garden.local.gardener.cloud"}},
+						Controller: gardencorev1beta1.IngressController{
+							Kind: "nginx",
+						},
+					},
+					Provider: operatorv1alpha1.Provider{
+						Zones: []string{"a", "b", "c"},
+					},
+					Settings: &operatorv1alpha1.Settings{
+						VerticalPodAutoscaler: &operatorv1alpha1.SettingVerticalPodAutoscaler{
+							Enabled: ptr.To(true),
+						},
+					},
+				},
+				VirtualCluster: operatorv1alpha1.VirtualCluster{
+					DNS: operatorv1alpha1.DNS{
+						Domains: []operatorv1alpha1.DNSDomain{{Name: "virtual-garden.local.gardener.cloud"}},
+					},
+					Gardener: operatorv1alpha1.Gardener{
+						ClusterIdentity: "test",
+					},
+					Kubernetes: operatorv1alpha1.Kubernetes{
+						Version: "1.31.1",
+					},
+					Maintenance: operatorv1alpha1.Maintenance{
+						TimeWindow: gardencorev1beta1.MaintenanceTimeWindow{
+							Begin: "220000+0100",
+							End:   "230000+0100",
+						},
+					},
+					Networking: operatorv1alpha1.Networking{
+						Services: []string{"100.64.0.0/13"},
+					},
+				},
+			},
+		}
+
+		By("Create Garden")
```

Comment:
Do you think it makes sense to add a test case when a `Garden` does not exist, just to check that the reconciler doesn't error and properly does not register the additional watches?